clinica/iotools/converters/adni_to_bids/adni_utils.py

```diff
@@ -1443,108 +1468,177 @@ def create_file(
+
+            # Conditions to check if output NIFTI files exists,
+            # and, if DWI, if .bvec and .bval files are also present
+            nifti_exists = (
+                file_without_extension.with_suffix(".nii").is_file()
+                or output_image.is_file()
             )
-            return None
-
-        # Case when JSON file was expected, but not generated by dcm2niix
-        elif (
-            generate_json == "y"
-            and not file_without_extension.with_suffix(".json").exists()
-        ):
-            cprint(
-                msg=f"JSON file not generated by dcm2niix for subject {subject} and session {session}",
-                lvl="warning",
+            dwi_bvec_and_bval_exist = not (modality == "dwi") or (
+                file_without_extension.with_suffix(".bvec").is_file()
+                and file_without_extension.with_suffix(".bval").is_file()
             )
 
-        if modality_specific[modality]["to_center"]:
-            output_image, error_msg = center_nifti_origin(
-                file_without_extension.with_suffix(".nii"), output_image
-            )
-            if error_msg:
-                cprint(msg=error_msg, lvl="error")
-                raise ValueError(error_msg)
-            file_without_extension.with_suffix(".nii").unlink()
+            # Check if conversion worked (output files exist)
+            if not nifti_exists or not dwi_bvec_and_bval_exist:
+                cprint(
+                    msg=f"Conversion with dcm2niix failed for subject {subject} and session {session}",
+                    lvl="warning",
+                )
+                return np.nan
 
-    else:
-        if modality_specific[modality]["to_center"]:
-            output_image, error_msg = center_nifti_origin(image_path, output_image)
-            if error_msg:
+            # Case when JSON file was expected, but not generated by dcm2niix
+            elif (
+                generate_json == "y"
+                and not file_without_extension.with_suffix(".json").exists()
+            ):
                 cprint(
-                    msg=(
-                        f"For subject {subject} in session {session}, "
-                        f"an error occurred whilst recentering Nifti image: {image_path}"
-                        f"The error is: {error_msg}"
-                    ),
-                    lvl="error",
+                    msg=f"JSON file not generated by dcm2niix for subject {subject} and session {session}",
+                    lvl="warning",
+                )
+
+            if modality_specific[modality]["to_center"]:
+                output_image, error_msg = center_nifti_origin(
+                    file_without_extension.with_suffix(".nii"), output_image
                 )
+                if error_msg:
+                    cprint(msg=error_msg, lvl="error")
+                    raise ValueError(error_msg)
+                file_without_extension.with_: wqsuffix(".nii").unlink()
```

Comment:
```suggestion
                file_without_extension.with_suffix(".nii").unlink()
```

This one is definitely my mistake but it should go back to this, so I would appreciate it if you made the modification

Reply:
Noted